src/openrct2/ride/RideRatings.cpp

```diff
@@ -816,11 +870,181 @@ static void ride_ratings_score_close_proximity(RideRatingUpdateState& state, Til
 
 static void RideRatingsCalculate(RideRatingUpdateState& state, Ride& ride)
 {
-    auto calcFunc = RideRatingsGetCalculateFunc(ride.type);
-    if (calcFunc != nullptr)
+    RideRatingsDescriptor rrd = ride.GetRideTypeDescriptor().RatingsData;
+
+    switch (rrd.Type)
+    {
+        case RatingsCalculationType::Normal:
+            if (!(ride.lifecycle_flags & RIDE_LIFECYCLE_TESTED))
+                return;
+            break;
+        case RatingsCalculationType::FlatRide:
+            ride.lifecycle_flags |= RIDE_LIFECYCLE_TESTED;
+            ride.lifecycle_flags |= RIDE_LIFECYCLE_NO_RAW_STATS;
+            break;
+        case RatingsCalculationType::Stall:
+            ride.upkeep_cost = ride_compute_upkeep(state, ride);
+            ride.window_invalidate_flags |= RIDE_INVALIDATE_RIDE_INCOME;
+            // Exit ratings
+            return;
+    }
+
+    ride.unreliability_factor = rrd.Unreliability;
+    SetUnreliabilityFactor(ride);
+
+    auto shelteredEighths = get_num_of_sheltered_eighths(ride);
+    ride.sheltered_eighths = (rrd.RideShelter == -1) ? shelteredEighths.TotalShelteredEighths : rrd.RideShelter;
+
+    RatingTuple ratings = rrd.BaseRatings;
+    // Apply Modifiers
+    for (const auto& modifier : rrd.Modifiers)
     {
-        calcFunc(ride, state);
+        switch (modifier.Type)
```

Comment:
I feel like we should store the function in modifier then we can avoid this huge switch here, the modifier as a type would be only relevant for when we import that one day from a json/object file.

Reply:
A few of the functions have slightly different parameters which kinda throws a wrench in that, would there be a way to avoid having to pass `shelteredEighths.TrackShelteredEighths` to every one of them? Would love to avoid the big stupid switch statement

Reply:
It looks like modifying RideRatingsUpdateState requires bumping the save version, so maybe best saved for later. And I'm not entirely sure what the deal is with the ride shelter code, I think it distinguishes between track shelter (tunnels) and vehicle shelter (enclosed cars) but not sure.